modules/API/API.js

```diff
@@ -177,13 +179,35 @@ class API {
          */
         this._enabled = true;
 
+        this._notifyLargeVideoVisibilityChanged
+            = this._notifyLargeVideoVisibilityChanged.bind(this);
+
         APP.conference.addListener(
             JitsiMeetConferenceEvents.DESKTOP_SHARING_ENABLED_CHANGED,
             onDesktopSharingEnabledChanged);
 
+        APP.UI.addListener(
+            UIEvents.LARGE_VIDEO_AVATAR_VISIBLE,
+            this._notifyLargeVideoVisibilityChanged);
+
```

Comment:
Why did you choose to add an event listener for ui events here? I ask because I noticed the pattern is something else (mainly conference.js) listens for ui event and then calls the api, but here the pattern is being changed so that api is listening directly. Do you feel whatever the reason is it's the better choice?

Reply:
I'm really not sure about this one. I was thinking the same thing. Since I'm still not sure which is the best way to do it, I would change it if you think it will be better to call _notifyLargeVideoVisibilityChanged from outside.

So do you think it will be OK if I call _notifyLargeVideoVisibilityChanged from VideoContainer.js? Or maybe add the listener somewhere else?

Reply:
I don't really see either as better, and the only marginal favor I currently see is in keeping consistency with the set pattern of conference calling the API. I just want to know why you preferred this approach.

Reply:
I don't have a strong reason for it. I was wondering where to execute it and at the end I added a listener. The benefit would be if we emit the event in multiple places but this is not the case for now. I'll move it to VideoContainer.js.

Reply:
It is moved now!